packages/dds/tree/src/codec/codec.ts

```diff
@@ -457,19 +452,21 @@ export enum FluidClientVersion {
 	/** Fluid Framework Client 2.52 and newer. */
 	// New formats introduced in 2.52:
 	// - DetachedFieldIndex FormatV2
-	v2_52 = 2.052,
+	v2_52: "2.52.0" as MinimumVersionForCollab,
 
 	/**
 	 * Enable unreleased and unfinished features.
 	 * @remarks
+	 * pkgVersion will always be the latest version, therefore the version that enables all features.
+	 *
 	 * Using this value can result in documents which can not be opened in future versions of the framework.
 	 * It can also result in data corruption by enabling unfinished features which may not handle all cases correctly.
 	 *
 	 * This can be used with specific APIs when the caller has knowledge of what specific features those APIs will be opted into with it.
 	 * This is useful for testing features before they are released, but should not be used in production code.
 	 */
-	EnableUnstableFeatures = Number.POSITIVE_INFINITY,
-}
+	EnableUnstableFeatures: pkgVersion as MinimumVersionForCollab,
+};
```

Comment:
you don't want this object being mutable, and you should type check it:


```suggestion
} as const satisfies Record<string, MinimumVersionForCollab>;
```